docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+happened.
+
+There is also a `SidechainModify` transaction used to change new account
+creation amounts and the reward amounts. Note: if the reward amount changes
+between the time a transaction is initiated the time the reward is collected,
+the old amount is used (as that is the amount the source account paid).
+
+
+## Witness Server
+
+A witness server is an independent server that helps provide proof that some
+event happened on either the mainchain or the sidechain. When they detect an
+event of interest, they use the `SidechainAddWitness` transaction to add their
+attestation that the event happened. When a quorum on signatures are collected on
+the ledger, the transaction predicated on that event happening is unlocked (and
+may be triggered automatically when the quorum of signatures is reached).
+Witness servers are independent from the servers that run the chains themselves.
+
+It is possible for a witness server to provide attestations for one chain only -
+and it is possible for the door account on the mainchain to have a different
+signer's list than the door account on the sidechain. The initial implementation
+of the witness server assumes it is providing attestation for both chains,
+however it is desirable to allow witness servers that only know about one of
+the chains.
+
+The current design envisions two models for how witness servers are used. In the
+first model, the servers are completely private. They submit transactions to the
+chains themselves and collect the rewards themselves. Allowing the servers to be
+private had the advantage of greatly reducing the attack surface on these
+servers. They won't have to deal with adversarial input to their RPC commands,
+and since their ip address will be unknown, it will be hard to mount an DOS
+attack.
+
+In the second model, the witness server monitors events on a chain, but does not
+submit their signatures themselves. Instead, another party will pay the
+attestation server for their signature (for example, through a subscription
+fee), and the attestation server allows that party to collect the signer's
+reward. The account that the signer's reward goes to is part of the message that
+the witness server signs. In this model, it is likely that the witness server
+only listens to events on one chain. This model allows for a single witness
+server to act as a witness for multiple sidechains.
+
+As a side note, since submitting a signature requires submitting a transaction
+and paying a fee, supporting rewards for signatures is an important requirement.
+Of course, the reward can be higher than the fee, providing an incentive to
+running a witness server.
+
+## Why use the signer's list on the account
+
+The signatures that the witness servers use must match the signatures on that
+door's signer's list. But this isn't the only way to implement this. A sidechain
+ledger object could contain a signer's list that's independent from the door
```

Comment:
Dumb question: if the door signer list is a better option as you suggest, do we need a signer list on the sidechain ledger object?

Reply:
No, that went away. It's no longer on side sidechain ledger object at all.

Reply:
sounds good